src/components/PolygonAddressInfo.js

```diff
@@ -0,0 +1,88 @@
+/* global IqonHash */
+/* global LoginFileConfig */
+/* global TemplateTags */
+/* global I18n */
+
+class PolygonAddressInfo { // eslint-disable-line no-unused-vars
+    /**
+     * @param {string} address
+     * @param {string} [label]
+     * @param {'none' | 'usdc' | 'unknown'} [logo = 'none']
+     */
+    constructor(address, label, logo = 'none') {
+        this._address = address;
+        this._label = label;
+        this._logo = logo;
+    }
+
+    /**
+     * Inserts this AddressInfo into $el overwriting the original content of $el.
+     * @param {HTMLElement} [$el]
+     * @returns {HTMLElement}
+     */
+    render($el) {
+        $el = $el || document.createElement('div');
+        $el.textContent = '';
+        $el.classList.add('polygon-address-info');
+
+        // Avatar or Icon
+        if (this._logo === 'none') {
+            const $avatar = document.createElement('div');
+            $avatar.classList.add('avatar');
+            if (this._label) {
+                $avatar.textContent = this._label.substring(0, 1).toUpperCase();
+                const color = IqonHash.getBackgroundColorIndex(this._label);
+                $avatar.classList.add(LoginFileConfig[color].className, 'initial');
+            } else {
+                $avatar.classList.add('unlabelled');
+            }
+            $el.appendChild($avatar);
+        } else if (this._logo === 'usdc') {
+            const $img = document.createElement('img');
+            $img.classList.add('logo');
+            $img.src = '../../assets/icons/usdc.svg';
+            $el.appendChild($img);
+        } else if (this._logo === 'unknown') {
+            const $img = document.createElement('img');
+            $img.classList.add('logo');
+            $img.src = '../../assets/icons/unknown.svg';
+            $el.appendChild($img);
+        }
+
+        // Label
+        const $label = document.createElement('div');
+        $label.classList.add('label');
+        if (this._label) {
+            $label.textContent = this._label;
+        } else {
+            $label.textContent = I18n.translatePhrase('label-unknown');
```

Comment:
```suggestion
            $label.textContent = I18n.translatePhrase('label-unknown');
            $label.dataset.i18n = 'label-unknown';
```
Also set i18n dataset to support reactive update on language switch.